go/vt/vtgate/buffer/buffer_helper_test.go

```diff
@@ -121,7 +121,7 @@ func waitForState(b *Buffer, want bufferState) error {
 func waitForPoolSlots(b *Buffer, want int) error {
 	start := time.Now()
 	for {
-		got := b.bufferSizeSema.Size()
+		got := b.bufferSize
```

Comment:
I'm not even sure what the old code means; the semaphore acts both as an atomic primitive _and_ indicates the size?

Reply:
We had code that indeed checked this size but the `x/sync` Go semaphore doesn't expose that (as it's a total weight, not a real size). 

That's why this had to be added as a separate field. Agreed it's a bit wonky but I didn't want to go and change behavior here.